### PR TITLE
Edit contact command

### LINKED ISSUE #225: [Contacts] Edit contact command should not allow duplicate phone and email

### PR DESCRIPTION
**Pull request by Chronoxy**
fixes #225 

Missing test cases. To be added

### CODE REVIEW

File: src/main/java/tutorease/address/logic/commands/EditContactCommand.java [hunk 1]
```diff
@@ -92,6 +94,14 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(MESSAGE_DUPLICATE_PERSON);
         }
 
+        if (editPersonDescriptor.email != null && model.hasSameEmail(editedPerson)) {
+            throw new CommandException(MESSAGE_DUPLICATE_EMAIL);
+        }
+
+        if (editPersonDescriptor.phone != null && model.hasSamePhone(editedPerson)) {
+            throw new CommandException(MESSAGE_DUPLICATE_PHONE);
+        }
```
Comment (NorbertLoh) [inline src/main/java/tutorease/address/logic/commands/EditContactCommand.java:103]: Perhaps can extract out these validation methods to increase SLAP.

**Review by NorbertLoh — COMMENTED** [review]
Nits to fix. Otherwise LGTM!

File: src/main/java/tutorease/address/logic/Messages.java [hunk 2]
```diff
@@ -22,6 +22,10 @@ public class Messages {
             "Multiple values specified for the following single-valued field(s): ";
     public static final String MESSAGE_NO_LESSONS_FOUND = "No lessons found!";
     public static final String MISSING_PREFIX = "%s prefix is missing!\n" + "%s";
+    public static final String MESSAGE_DUPLICATE_PHONE = "A person with this phone number already exists in the "
+            + "address book!";
+    public static final String MESSAGE_DUPLICATE_EMAIL = "A person with this email already exists in the "
+            + "address book!";
```
Comment (Ma-Xizi) [inline src/main/java/tutorease/address/logic/Messages.java:28]: Nice to incoporate standard messages into Messages!

**Review by Ma-Xizi — APPROVED** [review]

**Review by NorbertLoh — APPROVED** [review]
LGTM!

File: src/main/java/tutorease/address/logic/commands/EditContactCommand.java [hunk 3]
```diff
@@ -148,6 +148,24 @@ public String toString() {
                 .toString();
     }
 
+    /**
+    * Checks if the edited person would create a duplicate entry in the model based on unique attributes.
+    * This method verifies that the edited person's details do not conflict with existing persons in the model.
+    */
```
Comment (ethantyh692) [inline src/main/java/tutorease/address/logic/commands/EditContactCommand.java:154]: javadocs here can follow the format mentioned in the textbook with appropriate tags like @params @throws

File: src/test/java/tutorease/address/logic/commands/EditContactCommandTest.java [hunk 4]
```diff
@@ -226,4 +226,35 @@ public void execute_updateContact_guardian() throws ParseException {
...
+    }
+
+    @Test
+    public void execute_duplicatePhone_failure() {
+        // Arrange: Show only the first person in the filtered list
+        showPersonAtIndex(model, INDEX_FIRST_PERSON);
+
+        // Act: Edit the first person in the filtered list to have the same phone number as the second person
+        Person personInList = model.getTutorEase().getPersonList().get(INDEX_SECOND_PERSON.getZeroBased());
+        EditPersonDescriptor descriptor = new EditPersonDescriptor();
```
Comment (ethantyh692) [inline src/test/java/tutorease/address/logic/commands/EditContactCommandTest.java:251]: good use of comments to make it clear!

**Review by ethantyh692 — COMMENTED** [review]
Few Javadocs nits, besides that, LGTM

File: src/main/java/tutorease/address/logic/commands/EditContactCommand.java (hunk 3 above)
Comment (Chronoxy) [reply src/main/java/tutorease/address/logic/commands/EditContactCommand.java:165]: Fixed!

**Pull request merged**